spacetemplate/repository.go

```diff
@@ -0,0 +1,273 @@
+package spacetemplate
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormsupport"
+	"github.com/almighty/almighty-core/log"
+	"github.com/almighty/almighty-core/space"
+	"github.com/almighty/almighty-core/workitem"
+	"github.com/almighty/almighty-core/workitem/link"
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"golang.org/x/net/context"
+)
+
+// Repository describes interactions with space templates
+type Repository interface {
+	Create(ctx context.Context, st *SpaceTemplate) error
+	List(ctx context.Context) ([]SpaceTemplate, error)
+	Load(ctx context.Context, spaceTemplateID uuid.UUID) (*SpaceTemplate, error)
+	// Apply tries to load the space template id as stored in the space and then
+	// creates all the artifacts like work item type and work item link types
+	Apply(ctx context.Context, spaceTemplateID uuid.UUID, spaceID uuid.UUID) error
+}
+
+// NewRepository creates a new space template repository
+func NewRepository(db *gorm.DB) Repository {
+	return &GormRepository{db: db}
+}
+
+// GormRepository is the implementation of the repository interface for space
+// templates.
+type GormRepository struct {
+	db *gorm.DB
+}
+
+// Create creates a new space template.
+func (r *GormRepository) Create(ctx context.Context, st *SpaceTemplate) error {
+	defer goa.MeasureSince([]string{"goa", "db", "space_template", "create"}, time.Now())
+
+	if uuid.Equal(st.ID, uuid.Nil) {
+		st.ID = uuid.NewV4()
+	}
+
+	if err := st.Validate(); err != nil {
+		log.Info(ctx, map[string]interface{}{"space_template": st, "err": err}, "space template is invalid")
```

Comment:
I have an example: 

I have one user that is trying to access my application with a wrong token and he is getting a generic error. 

If I raise the logging level to `warning` in order to avoid the container to be chatty. I won't see that someone was trying to reach my application and use a wrong token. So I will need to re-deploy all the containers and change the logging level.

I think the purpose is helping ourselves to maintain the application and being able to easily troubleshoot any problem.